src/hotspot/share/gc/parallel/parallelScavengeHeap.cpp

```diff
@@ -540,44 +540,48 @@ void ParallelScavengeHeap::object_iterate(ObjectClosure* cl) {
 }
 
 // The HeapBlockClaimer is used during parallel iteration over the heap,
-// allowing workers to claim heap blocks, gaining exclusive rights to these blocks.
+// allowing workers to claim heap areas ("blocks"), gaining exclusive rights to these.
 // The eden and survivor spaces are treated as single blocks as it is hard to divide
 // these spaces.
-// The old spaces are divided into serveral fixed-size blocks.
+// The old space is divided into fixed-size blocks.
 class HeapBlockClaimer : public StackObj {
-  // Index of iterable block, negative values for indexes of young generation spaces,
-  // zero and positive values for indexes of blocks in old generation space.
-  ssize_t _claimed_index;
- public:
-  static const ssize_t EdenIndex = -2;
-  static const ssize_t SurvivorIndex = -1;
+  size_t _claimed_index;
+
+public:
+  static const size_t InvalidIndex = (size_t)SIZE_MAX;
+  static const size_t EdenIndex = 0;
+  static const size_t SurvivorIndex = 1;
+  static const size_t NumNonOldGenClaims = 2;
 
   HeapBlockClaimer() : _claimed_index(EdenIndex) { }
   // Claim the block and get the block index.
-  bool claim_and_get_block(ssize_t* block_index) {
-    assert(block_index != NULL, "Invalid index pointer");
-    *block_index = Atomic::fetch_and_add(&_claimed_index, 1);
-    ssize_t iterable_blocks = (ssize_t)ParallelScavengeHeap::heap()->old_gen()->iterable_blocks();
-    if (*block_index >= iterable_blocks) {
-      return false;
-    }
-    return true;
+  size_t claim_and_get_block() {
+    size_t block_index;
+    block_index = Atomic::fetch_and_add(&_claimed_index, (size_t)1);
```

Comment:
Instead of the cast please use 1u
```suggestion
    block_index = Atomic::fetch_and_add(&_claimed_index, 1u);
```